### PR TITLE
Implementation of component-test blueprint

### PR DESCRIPTION
**Pull request by alexander-alvarez**
<details>
<summary> Old description </summary>
I pretty much gutted https://github.com/emberjs/ember.js/tree/master/blueprints/component-test and adjusted to the best of my knowledge. 
I didn't know what to do for unit tests.

What we get from running `ember g component-test my-component -u` and then the codemod is:
```
import { module, test } from 'qunit';
import { setupTest } from 'ember-qunit';

module('Unit | Component | my component', function(hooks) {
  setupTest(hooks);

  test('it renders', function(assert) {
    
    // Creates the component instance
    /*let component =*/ this.owner.factoryFor('component:my-component').create();
    // Renders the component to the page
    this.render();
    assert.equal(this.$().text().trim(), '');
  });
});
```

- Do we want this? In that case, I can adjust the current implementation.

- As a general rule do we want the equivalent running the codemod on our current blueprints? or is now a time to change the format?

- Also what's the best way to test the blueprints?

**/cc @rwjblue @bendemboski @tbieniek**
</details> 

Related to https://github.com/emberjs/ember.js/issues/15933, I stole implementations and testcases from  https://github.com/emberjs/ember.js/pull/15934

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,12 +26,16 @@
     "ember-cli": "~2.16.2",
     "ember-cli-dependency-checker": "^2.0.0",
     "ember-cli-eslint": "^4.0.0",
+    "ember-cli-get-component-path-option": "^1.0.0",
```
Comment (rwjblue) [inline package.json]: These deps added need to be in `dependencies` section (not devDeps)

**Review by rwjblue — COMMENTED** [review]
I'll need to play around a bit with it to confirm the exact output, but this looks good!

**Comment by Turbo87** [comment]
@alexander-alvarez it would probably be good to use https://github.com/ember-cli/ember-cli-blueprint-test-helpers to test the output of the new blueprints

**Comment by alexander-alvarez** [comment]
note to self -- update to match implementation from https://github.com/emberjs/ember.js/pull/15934

**Comment by alexander-alvarez** [comment]
@Turbo87 updated with your changes that are in `ember.js#master` and test cases

**Review by rwjblue — COMMENTED** [review]
Seems good to me...

**Pull request merged**